src/data_readers/numpy_conduit_cache.cpp

```diff
@@ -0,0 +1,70 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2016, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#include "lbann/data_readers/numpy_conduit_cache.hpp"
+#include "lbann/utils/exception.hpp"
+
+#ifdef LBANN_HAS_CONDUIT
+
+namespace lbann {
+
+
+void numpy_conduit_cache::load(const std::string filename, int data_id) {
+
+  try {
+    m_numpy[data_id] = cnpy::npz_load(filename);
+    conduit::Node &n = m_data[data_id];
+    std::map<std::string, cnpy::NpyArray> &a = m_numpy[data_id];
```

Comment:
Related to my comment about packed conduit nodes, we may want to have ownership of the array transfer to conduit.